pkg/utils/remediation_diff.go

```diff
@@ -1,12 +1,237 @@
+	fmt.Println(prCtx.inconsistent)
+}
+
+// ParseResultContext.addParsedResults add a subsequent batch of results that must be examined
+// for consistency
+func (prCtx *ParseResultContext) addParsedResults(source string, newResults []*ParseResult) {
+	for _, consistentResult := range prCtx.consistent {
+		consistentResult.processed = false
+	}
+
+	for _, pr := range newResults {
+		consistentPr, ok := prCtx.consistent[pr.Id]
+		if !ok {
+			// this batch has an extra item, save it as a diff with (only so far) this source
+			prCtx.addInconsistentResult(pr.Id, pr, source)
+			continue
+		}
+		consistentPr.processed = true
+
+		ok = diffChecks(consistentPr.CheckResult, pr.CheckResult)
+		if !ok {
+			// remove the check from consistent, add it to diff, but TWICE
+			// once for the sources from the consistent list and once for the new source
+			prCtx.addInconsistentResult(pr.Id, &consistentPr.ParseResult, consistentPr.sources...)
+			delete(prCtx.consistent, pr.Id)
+			prCtx.addInconsistentResult(pr.Id, pr, source)
+			continue
+		}
+
+		// OK, same as a previous result in consistent, just append the source
+		consistentPr.sources = append(consistentPr.sources, source)
+	}
+
+	// Make sure all previously consistent items were touched, IOW we didn't receive
+	// fewer items by moving all previously untouched items to the inconsistent list
+	for _, consistentResult := range prCtx.consistent {
+		if consistentResult.processed == true {
+			continue
+		}
+		// Deleting an item from a map while iterating over it is safe, see https://golang.org/doc/effective_go.html#for
+		prCtx.addInconsistentResult(consistentResult.Id, &consistentResult.ParseResult, consistentResult.sources...)
+		delete(prCtx.consistent, consistentResult.Id)
+	}
+}
+
+// ParseResultContext.ReconcileInconsistentResults interates through all inconsistent results
+// and tries to reconcile them, creating a single consistent ParseResultContextItem for each
+func (prCtx *ParseResultContext) reconcileInconsistentResults() error {
+	for id, inconsistentResultList := range prCtx.inconsistent {
+		fmt.Printf("Reconciling rule ID %s\n", id)
+
+		if len(inconsistentResultList) < 1 {
+			fmt.Printf("Nothing to reconcile")
+			continue
+		}
+
+		reconciled := reconcileInconsistentResult(inconsistentResultList)
+		if _, ok := prCtx.consistent[id]; ok {
+			// FIXME: Instead of returning an error here, create a consistent check with
+			// an ERROR
+			return fmt.Errorf("BUG: a rule %s can't be consistent an inconsistent at the same time", id)
+		}
+		prCtx.consistent[id] = reconciled
+	}
+
+	return nil
+}
+
+func (prCtx *ParseResultContext) GetConsistentResults() ([]*ParseResultContextItem, error) {
+	err := prCtx.reconcileInconsistentResults()
+	if err != nil {
+		return nil, err
+	}
+
+	consistentList := make([]*ParseResultContextItem, len(prCtx.consistent))
+	idx := 0
+
+	for _, item := range prCtx.consistent {
+		consistentList[idx] = item
+		idx++
```

Comment:
why add them with the index if you could just append them?

```
consistentList = append(consistentList, item)
```

Reply:
I don't honestly know why I wrote the code the way I did. Fixed.